### PR TITLE
chore: Add migration guide

### PR DESCRIPTION
**Pull request by sfc-gh-jcieslak**
add migration guide / notes for future api changes

### CODE REVIEW

**Review by sfc-gh-swinkler — COMMENTED** [review]
Can we at least paste in the documentation for upgrading form chanzuckerber provider? Otherwise this just feels empty and pointless.

File: MIGRATION_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Migration guide
+
+This document is meant to help you migrate your Terraform config to new newest version
```
[the remark this answers is not in the document]
Comment (sfc-gh-jcieslak) [reply MIGRATION_GUIDE.md:3]: I guess the one Scott mentioned, and we recently have changed some of the things to the provider, so I'm guessing yes. I'll try to gather some info about what could go in here and then will update you to check again.

File: MIGRATION_GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    token_endpoint  = "<endpoint>"
+    redirect_uri    = "<redirect_uri>"
+  }
+}
+```
+
+#### *(remove redundant information)* region
+
+Specifying a region is a legacy thing and according to https://docs.snowflake.com/en/user-guide/admin-account-identifier
+you can specify a region as a part of account parameter. Specifying account parameter with the region is also condiered legacy,
```
Comment (sfc-gh-asawicki) [inline MIGRATION_GUIDE.md:55]: condiered -> considered

**Review by sfc-gh-asawicki — APPROVED** [review]

File: MIGRATION_GUIDE.md (hunk 2 above)
Comment (sfc-gh-jcieslak) [reply MIGRATION_GUIDE.md]: Fixed

**Review by sfc-gh-swinkler — APPROVED** [review]

**Review by sfc-gh-asawicki — APPROVED** [review]

**Pull request merged**